#!/usr/bin/env python3
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("P_xy", double_t, 0, "P_xy",    2.5, 0,   10.0)
gen.add("I_xy", double_t, 0, "I_xy",    0.0, 0,   2.0)
gen.add("D_xy", double_t, 0, "D_xy",    0.35, 0,   2.0)
gen.add("P_z", double_t, 0, "P_z",    5.0, 0,   10.0)
gen.add("I_z", double_t, 0, "I_z",    0.0, 0,   2.0)
gen.add("D_z", double_t, 0, "D_z",    0.0, 0,   2.0)
gen.add("P_yaw", double_t, 0, "P_yaw",    1.0, 0,   10.0)
gen.add("vmax_xyz", double_t, 0, "vmax_xyz",    1.0, 0,   10.0)
gen.add("vmax_yaw", double_t, 0, "vmax_yaw",    0.5, 0,   2.0)

exit(gen.generate("tricopter", "tricopter", "PID"))